install/helm/open-match/subcharts/open-match-customize/templates/_helpers.tpl

```diff
@@ -0,0 +1,40 @@
+{*
+ Copyright 2019 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*}
+
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+Instead of .Chart.Name, we hard-code "open-match" as we need to call this from subcharts, but get the
+same result as if called from this chart.
+*/}}
+{{- define "openmatch.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default "open-match" .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "openmatch.function.hostName" -}}
+{{- .Values.function.hostName | default (printf "%s-function" (include "openmatch.fullname" . ) ) -}}
+{{- end -}}
```

Comment:
Why `openmatch.function.hostName` is defined here while `openmatch.evaluator.hostName` is defined in  `install/helm/open-match/templates/_helpers.tpl`. Both of them are customizable components and should be at the same level.

Reply:
`openmatch.evaluator.hostName` is also used in `om-configmap-override.yaml` out of the sub-chart. Defining it in the parent chart means that this sub-chart can not be installed independently. But it may not be an issue.

Reply:
`openmatch.function.hostName` is defined here because it's only _used_ here, while, as you have noted, `openmatch.evaluator.hostName` is also used (once) in the parent chart. See #1244. The configuration for evaluator hostname (and matchFunction hostname) are still done in the top-level chart's values.yaml.

I probably should have renamed `openmatch.function.hostName` to `openmatchcustomize.function.hostName`, on consideration.

I'll fix that, and I'll make sure that the subcharts don't carry straight-up copies of the parent chart templates, since I haven't replicated all the used named templates down into the subcharts.

The structure of the charts means that the sub-charts are a long way from being independently installed, as there is still a log of dependence in the subcharts on parent chart named templates. I didn't try to address this here, except when it was easy, and I'm not sure it's a goal to have them independently installable.

I suggested in #1244 rolling the sub-charts back into the main chart, partly for this reason, because it sends a slightly weird signal on first glance.